src/variable_rate.jl

```diff
@@ -281,20 +286,34 @@ mutable struct VR_DirectEventCache{T, RNG <: AbstractRNG, F1, F2}
     rate_funcs::F1
     affect_funcs::F2
     cum_rate_sum::Vector{T}
+end
 
-    function VR_DirectEventCache(jumps::JumpSet, ::Type{T}; rng = DEFAULT_RNG) where T
-        initial_threshold = randexp(rng, T)
-        vjumps = jumps.variable_jumps
+function VR_DirectEventCache(jumps::JumpSet, ::VR_Direct, prob, ::Type{T}; rng = DEFAULT_RNG) where T
+    initial_threshold = randexp(rng, T)
+    vjumps = jumps.variable_jumps
 
-        # handle vjumps using tuples
-        rate_funcs, affect_funcs = get_jump_info_tuples(vjumps)
+    # handle vjumps using tuples
+    rate_funcs, affect_funcs = get_jump_info_tuples(vjumps)
         
-        cum_rate_sum = Vector{T}(undef, length(vjumps))
+    cum_rate_sum = Vector{T}(undef, length(vjumps))
         
-        new{T, typeof(rng), typeof(rate_funcs), typeof(affect_funcs)}(zero(T), 
-            initial_threshold, zero(T), initial_threshold, zero(T), rng, rate_funcs, 
-            affect_funcs, cum_rate_sum)
-    end
+    VR_DirectEventCache{T, typeof(rng), typeof(rate_funcs), typeof(affect_funcs)}(zero(T), 
+        initial_threshold, zero(T), initial_threshold, zero(T), rng, rate_funcs, 
+        affect_funcs, cum_rate_sum)
+end
+
+function VR_DirectEventCache(jumps::JumpSet, ::VR_DirectFW, prob, ::Type{T}; rng = DEFAULT_RNG) where T
+    initial_threshold = randexp(rng, T)
+    vjumps = jumps.variable_jumps
+
+    # handle vjumps using tuples
+    rate_funcs, affect_funcs = get_jump_info_vr_fwrappers(vjumps, prob)
+        
+    cum_rate_sum = Vector{T}(undef, length(vjumps))
+        
+    VR_DirectEventCache{T, typeof(rng), typeof(rate_funcs), typeof(affect_funcs)}(zero(T), 
```

Comment:
```suggestion
    VR_DirectEventCache{T, typeof(rng), typeof(rate_funcs), Any}(zero(T), 
```
The type of the affect vector  in the cache needs to be `Any`. Later we will concretize it to the specific type when we want to actually call an affect.